### PR TITLE
AWS SDK Go V2 Migration

### PR DESCRIPTION
**Pull request by fredjoonpark**
*Issue #, if available:*

---

*Description of changes:*

This change upgrades the underlying AWS Go SDK from V1 to V2.

Some notes and highlights regarding significant changes in V2 for your reference:
- V2 SDK splits services into separate modules
- The config module in V2 introduces a single entry point (`config.LoadDefaultConfig()`) which consolidates loading AWS credentials, regions, etc
- service client APIs have been refined for more idiomatic Go usage (ie. relying less on `aws.String()` wrappers and providing typed fields where possible)
- all SDK operations now require a `context.Context` (supporting graceful cancellation, timeouts etc)
- the internal request pipeline has been restructured around `Smithy` middleware. Errors now follow a more consistent structure influenced by Smithy’s model-driven error definitions, making it easier to identify and handle specific error conditions (like `ResourceNotFound` or `AccessDenied`) through typed errors rather than relying on string matching.
- V2 SDK now uses a [dedicated paginator](https://aws.github.io/aws-sdk-go-v2/docs/migrating/#paginators) with an iterative interface (`HasMorePages()` and `NextPage()`) to handle multi-page API responses. In our context, `NextPage` performs the network call for fetching query responses (previously `QueryPages`). 
- Linted with `goimports`, let me know if you guys have a particular preference or style.
- Added a `docker-compose.yml` as I find it useful, but I can remove it if it clutters the repo.

Additional resources:
- https://github.com/aws/aws-sdk-go-v2
- https://aws.github.io/aws-sdk-go-v2/docs/migrating/

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: integration/integration_test.go [hunk 1]
```diff
@@ -90,45 +101,56 @@ func TestWriteClient(t *testing.T) {
...
 	}
-	for _, test := range invalidTestCase {
-		t.Run(test.testName, func(t *testing.T) {
-			err := clientEnableFailOnLongLabelName.WriteClient().Write(test.request, test.credentials)
+
+	for _, tc := range invalidTestCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var client *timestream.Client
+			if tc.allowLongLabel {
+				client = createClient(t, logger, database, table, tc.creds, true, false)
```
Comment (forestmvey) [inline integration/integration_test.go:141]: Using one or two letter variables is not very readable.

File: integration/tls/tls_test.go [hunk 2]
```diff
@@ -217,12 +223,16 @@ func connectorStatusCheck(t *testing.T, dockerClient *client.Client, ctx context
...
-	sess, err := session.NewSession(&aws.Config{Region: aws.String(region)})
+	ctx := context.Background()
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(region))
 	require.NoError(t, err)
-	querySvc := timestreamquery.New(sess)
-	out, err := querySvc.Query(queryInput)
+
+	querySvc := timestreamquery.NewFromConfig(cfg)
+	queryInput := &timestreamquery.QueryInput{
+		QueryString: aws.String(fmt.Sprintf("SELECT count(*) from %s.%s", database, table)),
```
Comment (forestmvey) [inline integration/tls/tls_test.go:232]: Quotes for database and table

File: main.go [hunk 3]
```diff
@@ -108,15 +119,22 @@ func main() {
...
+		ctx := context.Background()
+		awsQueryConfigs, err := cfg.buildAWSConfig(ctx, cfg.maxRetries)
+		if err != nil {
+			timestream.LogError(logger, "Failed to build AWS configuration for query", err)
+			os.Exit(1)
+		}
 
-		awsQueryConfigs.MaxRetries = aws.Int(cfg.maxRetries)
-		timestreamClient.NewQueryClient(logger, awsQueryConfigs)
+		awsWriteConfigs, err := cfg.buildAWSConfig(ctx, writeClientMaxRetries)
```
Comment (forestmvey) [inline main.go:130]: Is there any reason the writeClientMaxRetries is hard-coded but the maxRetries isn't for the query client? If not I think we should add a flag and make query and write retries configurable.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+services:
```
Comment (forestmvey) [inline docker-compose.yml:1]: Having a docker-compose is nice. Can you add a section on how to run the connector using it in the README.

**Comment by trevorbonas** [comment]
Overall, looks good to me, once the other comments have been addressed.

**Review by trevorbonas — APPROVED** [review]
Looks good!

- [X] Unit tests passed.
- [X] Integration tests passed.
- [X] TLS tests passed.
- [X] Correctness tests passed.

**Review by forestmvey — APPROVED** [review]
Looks good. I validated all tests are passing and deployed the connector to test ingestion and remote_read. Can you add a note in the PR description about the returned status code changes for `badRequest` and `statusConflict`.

**Pull request merged**